docs/deployment.md

```diff
@@ -27,225 +28,301 @@ Configuration of the PDP is accomplished through two mechanisms:
 
-When you have completed a cycle of development and testing, you may wish to stop the Docker containers.
+In production, the frontend and backend run in separate WSGI containers. 
+This is because the front end serves short, non-blocking requests, whereas 
+the back end serves fewer long, process blocking requests.
 
-```
-docker-compose -f docker/dev-local/docker-compose.yaml down
-```
+### Docker
 
-***9 - Extra: Run an interactive bash shell inside a container***
+We deploy in Docker containers. All the Docker infrastructure necessary to
+construct a production deployment is found in 
+[`docker/production`](docker/production). The alert reader will note that
+this is very similar to the `docker/local-run` infrastructure.
 
-When the containers are running, you can poke around inside them and/or execute tests inside them by connecting to them interactively. In order to do this:
+In your deployment directory, you will need to:
 
-```
-docker exec -it <container> bash
-```
+1. Copy the contents of `docker/production/`.
 
-This starts a bash shell inside the container and connects you to it. You should see a command prompt like:
+1. Create `docker/production/common-with-passwords.env` from 
+`docker/local-run/common.env` by adding passwords for the `pcic_meta` and 
+`crmp` databases.
 
-```
-root@e320e9c22200:~/pdp# 
-```
+1. Create `docker/production/pgbounce_users-with-passwords.txt` from  
+`docker/dev-local/pgbounce_users.txt` by inserting correct md5 sums.
 
-Again, since your local codebase is mounted to the containers and installed in editable/development mode (`pip install -e .`), any code changes you make are eflected "live" inside the container, and so you may modify code externally and run the tests inside the container.
+Starting and stopping the containers is done in the usual fashion with
+`docker-compose`.
 
-TODO: Figure out why tests are trying to import from `/codebase` rather than from `/root/pdp`, or else use workdir `/codebase`. Geez.
+#### Testing a production deployment locally
 
-#### Notes
+When making changes to the production Docker infrastructure, it can be helpful
+to deploy the production containers locally. 
 
-1. Data files need only be mounted to the *backend* service.
-1. Oh yeah, you'll need to mount the gluster `/storage` volume to locally to `/storage` so that those data files are accessible.
-1. JS configuration files need only be mounted to the *frontend* service. An example one is included in this directory, and mounted. It overrides the default one in the project.
+To support this, we have also
+defined the file `docker/production/docker-compose-local.yaml`. Like
+`docker/local-run/docker-compose.yaml`, it starts a (different) set of
+containers that enable PDP to be run locally, in this case the production
+instance. Unfortunately, for reasons not entirely understood just yet,
+they too must be deployed to port 5000, i.e., `pdp.localhost:5000`,
```

Comment:
Is this port hard coded somewhere? This seems odd to me.

Reply:
The port is hardcoded in the docker-compose, but that's not really the issue. *Some* port has to be used, I picked 5000. But trying to use a different port, 5001, caused errors that I don't have the time to dig into. It *might* be another instance of varnish keeping around outdated access-control headers, but it might not either. Time ...